post_processing/orac_postproc.F90

```diff
@@ -672,95 +648,88 @@ subroutine orac_postproc(mytask, ntasks, lower_bound, upper_bound, &
                       input_primary(0)%cldtype(i,j,1) = SWITCHED_TO_WATER_TYPE
                    end if
                 end if
-
+  
                 ! Once phase is selected fill in the values 1st for wat then for
                 ! ice.
-
-                if (phase_flag == 1_byte) then
-                   call copy_class_specific_inputs(i, j, loop_ind(IWat), &
-                        input_primary(0), input_primary(IWat), &
-                        input_secondary(0), input_secondary(IWat), do_secondary)
+                ! Now fill in the phase values in the data structure
+                if (phase_flag == 1_byte) then ! Liquid water
                    input_primary(0)%phase(i,j) = IPhaseWat
-                else if (phase_flag == 2_byte) then
-                   call copy_class_specific_inputs(i, j, loop_ind(IIce), &
-                        input_primary(0), input_primary(IIce), &
-                        input_secondary(0), input_secondary(IIce), do_secondary)
+                else if (phase_flag == 2_byte) then ! Ice
                    input_primary(0)%phase(i,j) = IPhaseIce
-                else if (phase_flag == 3_byte) then
-                   ! Multilayer cloud
-                   call copy_class_specific_inputs(i, j, loop_ind(IMul), &
-                        input_primary(0), input_primary(IMul),&
-                        input_secondary(0), input_secondary(IMul), do_secondary)
+                else if (phase_flag == 3_byte) then ! Multilayer cloud
                    input_primary(0)%phase(i,j) = IPhaseMul
                 end if
 
-                ! Overwrite cc_total with cldmask for Cloud CCI
-                input_primary(0)%cc_total(i,j) = input_primary(0)%cldmask(i,j,1)
-                input_primary(0)%cc_total_uncertainty(i,j) = &
-                     input_primary(0)%cldmask_uncertainty(i,j,1)
-
-                if (input_primary(0)%cc_total(i,j) == 0.0) then
-                   ! Set phase to clear/unknown
-                   input_primary(0)%phase(i,j) = IPhaseClU
+                ! Now apply cloud_cci specific changes
+                if (cloud_cci) then
+                   ! Overwrite cc_total with cldmask
+                   input_primary(0)%cc_total(i,j) = input_primary(0)%cldmask(i,j,1)
+                   input_primary(0)%cc_total_uncertainty(i,j) = &
+                        input_primary(0)%cldmask_uncertainty(i,j,1)
+                   ! Set non-cloud pixels to Pavalonis clear/unknown phase, if
+                   ! they don't already have an aerosol phase assigned.
+                   if (input_primary(0)%cldmask(i,j,1) .eq. 0 .and. &
```

Comment:
I don't remember why we do this. Oli implemented it.

Reply:
I guess it is because cc_total (the cloud fraction) is always either 1 or 0, as is determined by the cloud mask (?). Anyway, it might as well be left as it is for this pull request, so as to be consistent. Possibly something to think about changing separately though.